helpers/kernel_config.go

```diff
@@ -88,6 +89,12 @@ const (
 	CONFIG_BPF_PRELOAD_UMD
 )
 
+// Constants declared here are used in the eBPF objects
+
+const (
+	CONFIG_ARCH_HAS_SYSCALL_WRAPPER KernelConfigOption = iota + 1000
```

Comment:
Why do we need this constant?

Reply:
To get it's value from kconfig file just as any other constant, in the same logic. Check Tracee PR to see I'm reading this kconfig variable value using this constant. We can extend this and I wanted to differentiate for the consts we use to check supportability.

Reply:
I see.
So, anytime we (or any other user of libbpfgo) will want to add some new config options (and there might be many of these), we will have to update this KernelConfigOption enum? I don't think that it will scale well...

When thinking about it, the KernelConfigOption values have no meaning, just an enumeration, right? So my question is, why do we even need this enum to be defined as int32? Why not defining it as a string? Anyways we convert it from string and to string, but do we really need these conversions? The natural type for this value would be string, am I wrong? And if we only need to make these conversions to int just so we can use it as a key to a bpf map, then I think that the responsibility of making such conversions should be of the user of libbpfgo (e.g. tracee), and not of libbpfgo itself. WDYT?

Reply:
Yes makes sense. The user (Tracee) is the one to be in charge of an array of map[uint32]strings that shall be used as source for the kconfig map - or something similar - so we don't have to specify inside the library but yet have it defined to kconfig_map.. let me work on that..

Reply:
@yanivagman done. Please check the new version together with https://github.com/aquasecurity/tracee/pull/981